ndcube/mixins/plotting.py

```diff
@@ -406,21 +409,32 @@ def _animate_cube_1D(self, plot_axis_index=-1, axes_coordinates=None,
                           ylabel="Data [{0}]".format(data_unit), **kwargs)
         return ax
 
-    def _derive_axes_coordinates(self, axes_coordinates, axes_units):
+    def _derive_axes_coordinates(self, axes_coordinates, axes_units, data_shape, edges=False):
         new_axes_coordinates = []
         new_axes_units = []
         default_labels = []
         default_label_text = ""
         for i, axis_coordinate in enumerate(axes_coordinates):
             # If axis coordinate is None, derive axis values from WCS.
             if axis_coordinate is None:
-                # N.B. This assumes axes are independent.  Fix this before merging!!!
-                new_axis_coordinate = self.axis_world_coords(i)
-                axis_label_text = self.world_axis_physical_types[i]
+                # Fix: We would downscale the dependent data into the shape of the axes.
+                xdata = self.axis_world_coords(i, edges=edges)
+                if xdata.ndim != data_shape[i]:
+                    axis_label_text = self.world_axis_physical_types[i]
+
+                    index = utils.wcs.get_dependent_data_axes(self.wcs, i, self.missing_axes)
+                    reduce_axis = np.where(index == np.array(i))[0]
```

Comment:
I don't think it's necessary to wrap ```i``` into a numpy array...?

Reply:
Sure, it was more of a defensive approach, would change it!

Reply:
Yeah, now I remember, since `index` is a numpy array, so for comparing the same objects, I have to convert `i` into a numpy array.